main.go

```diff
@@ -227,26 +231,50 @@ func main() {
 
 	log.V(2).Info("Configured the watch namespace", "watchNamespace", watchNamespace)
 
+	configPolicy := &policyv1.ConfigurationPolicy{}
+	secret := &corev1.Secret{}
+	operatorPolicy := &policyv1beta1.OperatorPolicy{}
+
 	if watchNamespace != "" {
-		cacheByObject[&policyv1.ConfigurationPolicy{}] = cache.ByObject{
-			Field: fields.SelectorFromSet(fields.Set{
-				"metadata.namespace": watchNamespace,
-			}),
+		cacheByObject[configPolicy] = cache.ByObject{
+			Namespaces: map[string]cache.Config{
+				watchNamespace: {},
+			},
 		}
 
-		cacheByObject[&corev1.Secret{}] = cache.ByObject{
-			Field: fields.SelectorFromSet(fields.Set{
-				"metadata.namespace": watchNamespace,
-				"metadata.name":      "policy-encryption-key",
-			}),
+		cacheByObject[secret] = cache.ByObject{
+			Namespaces: map[string]cache.Config{
+				watchNamespace: {
+					FieldSelector: fields.SelectorFromSet(fields.Set{
+						"metadata.name": "policy-encryption-key",
+					}),
+				},
+			},
 		}
 
 		if opts.enableOperatorPolicy {
-			cacheByObject[&policyv1beta1.OperatorPolicy{}] = cache.ByObject{
-				Field: fields.SelectorFromSet(fields.Set{
-					"metadata.namespace": watchNamespace,
+			cacheByObject[operatorPolicy] = cache.ByObject{
+				Namespaces: map[string]cache.Config{
+					watchNamespace: {},
+				},
+			}
+		}
+
+		if opts.targetKubeConfig == "" {
+			// ocmPolicyNs is cached only in non-hosted=mode
+			cacheByObject[configPolicy].
+				Namespaces[ocmPolicyNs] = cache.Config{}
+
+			cacheByObject[secret].Namespaces[ocmPolicyNs] = cache.Config{
+				FieldSelector: fields.SelectorFromSet(fields.Set{
+					"metadata.name": "policy-encryption-key",
 				}),
 			}
```

Comment:
I think this can be removed--I don't think the encryption secret is relevant for standalone policies.

Reply:
Thanks a lot!! I need this advice